src/components/Badge/Badge.tsx

```diff
@@ -13,14 +13,41 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import styled from '@emotion/styled';
+import React, {
+  forwardRef,
+  MouseEvent,
+  KeyboardEvent,
+  Ref,
+  HTMLProps
+} from 'react';
 import { css } from '@emotion/core';
 import isPropValid from '@emotion/is-prop-valid';
 
+import styled, { StyleProps } from '../../styles/styled';
 import { subHeadingKilo, focusOutline } from '../../styles/style-helpers';
 
+type OnClick = (
+  event: MouseEvent<HTMLDivElement> | KeyboardEvent<HTMLDivElement>
+) => void;
+type RefType = Ref<HTMLDivElement>;
+
+export interface BadgeProps extends HTMLProps<HTMLDivElement> {
+  /**
+   * Callback for the click event.
+   */
+  onClick?: OnClick;
+  /**
+   * Ensures text is centered and the badge looks like a circle.
+   */
+  circle?: boolean;
+  color?: 'neutral' | 'primary' | 'success' | 'warning' | 'danger';
+  /**
+   * The ref to the html div DOM element
+   */
+  ref?: RefType;
+  as?: string;
+}
+
 const COLOR_MAP = {
```

Comment:
Just a note, unrelated to this PR: this should probably change eventually, because we're hardcoding a mapping between a semantic color ("primary") to an actual color (blue). This prevents theme customization, "primary" should map to whatever color was set as primary.

The color values for primary:hover, primary:active, and so on should probably also be in the theme. Other thoughts?

cc @connor-baer

Reply:
Good point about the theme values for the primary colors, so...

`b500` → `p500`
`b700` → `p700`
`b900` → `p900`